lib/ocpp/v201/smart_charging.cpp

```diff
@@ -224,4 +339,41 @@ std::vector<ChargingProfile> SmartChargingHandler::get_station_wide_tx_default_p
     return station_wide_tx_default_profiles;
 }
 
+bool SmartChargingHandler::is_overlapping_validity_period(int candidate_evse_id,
+                                                          const ChargingProfile& candidate_profile) const {
+
+    if (candidate_profile.chargingProfilePurpose == ChargingProfilePurposeEnum::TxProfile) {
+        // This only applies to non TxProfile types.
+        return false;
+    }
+
+    auto conflicts_with = [candidate_evse_id, &candidate_profile](
+                              const std::pair<int32_t, std::vector<ChargingProfile>>& existing_profiles) {
+        auto existing_evse_id = existing_profiles.first;
+        if (existing_evse_id == candidate_evse_id) {
+            return std::any_of(existing_profiles.second.begin(), existing_profiles.second.end(),
+                               [&candidate_profile](const ChargingProfile& existing_profile) {
+                                   if (existing_profile.stackLevel == candidate_profile.stackLevel &&
+                                       existing_profile.chargingProfileKind == candidate_profile.chargingProfileKind &&
+                                       existing_profile.id != candidate_profile.id) {
+
+                                       return candidate_profile.validFrom <= existing_profile.validTo &&
+                                              candidate_profile.validTo >= existing_profile.validFrom; // reject
+                                   }
+                                   return false;
+                               });
+        }
+        return false;
+    };
+
+    return std::any_of(charging_profiles.begin(), charging_profiles.end(), conflicts_with);
+}
+
+void SmartChargingHandler::conform_validity_periods(ChargingProfile& profile) const {
+    profile.validFrom =
+        profile.validFrom.has_value() ? profile.validFrom.value() : ocpp::DateTime(date::utc_clock::now());
```

Comment:
Please replace these by either:
```
profile.validFrom = profile.validFrom.value_or(ocpp::DateTime()));
```
or
```
if (!profile.validFrom.has_value()) {
    profile.validFrom = ocpp::DateTime();
}
```

To me that would be more clear what the intention is. Also `DateTime()` already default constructs to `date::utc_clock::now()` ;)